thirdparty/kyaml/runfn/runfn.go

```diff
@@ -404,6 +404,9 @@ func (r *RunFns) getFunctionConfig() (*yaml.RNode, error) {
 	if len(nodes) > 1 {
 		return nil, fmt.Errorf("function config file '%s' must not contain more than 1 config", r.FnConfigPath)
 	}
+	if nodes[0].GetName() == "" {
+		return nil, fmt.Errorf("function config file '%s' is missing field `metadata.name`", r.FnConfigPath)
```

Comment:
See previous comment about wording.

Also, I am wondering if current code allows this validation to be be done in a common place for `eval and render`. Not a blocker if it doesn't.

Reply:
Looked back at a previous conversation and you mentioned making a method isKRM() which returns a validation error. I will try to do that instead.

Reply:
updated to share validation btwn render and eval